File: src/main/kotlin/com/yoloo/server/core/user/LocaleTranslatorFactory.java
package com.yoloo.server.core.user;

import com.googlecode.objectify.impl.Path;
import com.googlecode.objectify.impl.translate.*;

import java.util.Locale;

public class LocaleTranslatorFactory extends ValueTranslatorFactory<Locale, String> {

  /**
   * Create a new instance.
   */
  public LocaleTranslatorFactory() {
    super(Locale.class);
  }

  @Override
  protected ValueTranslator<Locale, String> createValueTranslator(TypeKey<Locale> tk, CreateContext ctx, Path path) {
    return new ValueTranslator<Locale, String>(String.class) {
      @Override
      protected Locale loadValue(String value, LoadContext ctx, Path path) throws SkipException {
        return Locale.forLanguageTag(value);
      }

      @Override
      protected String saveValue(Locale value, boolean index, SaveContext ctx, Path path) throws SkipException {
        return value.getLanguage();
      }
    };
  }
}

File: src/main/kotlin/com/yoloo/server/core/user/domain/objectify/translate/LocaleTranslatorFactory.kt
package com.yoloo.server.core.user.domain.objectify.translate

import com.googlecode.objectify.impl.Path
import com.googlecode.objectify.impl.translate.*
import java.util.*

/**
 * Create a new instance.
 */
class LocaleTranslatorFactory : ValueTranslatorFactory<Locale, String>(Locale::class.java) {

    override fun createValueTranslator(
        tk: TypeKey<Locale>,
        ctx: CreateContext,
        path: Path
    ): ValueTranslator<Locale, String> {
        return object : ValueTranslator<Locale, String>(String::class.java) {
            @Throws(SkipException::class)
            override fun loadValue(value: String, ctx: LoadContext, path: Path): Locale {
                return Locale.forLanguageTag(value)
            }

            @Throws(SkipException::class)
            override fun saveValue(value: Locale, index: Boolean, ctx: SaveContext, path: Path): String {
                return value.toLanguageTag()
            }
        }
    }
}
